docs/application/native/guides/text-input/autofill.md

```diff
@@ -0,0 +1,365 @@
+               break;
+           default:
+               break;
+   }
+
+   void
+   connect_autofill_daemon(autofill_h amh)
+   {
+       int ret;
+       ret = autofill_connect(amh, connection_status_changed_cb, NULL);
+       if (ret != AUTOFILL_ERROR_NONEt)
+           /* Error handling */
+   }
+   ```
+
+   You can use the autofill functions after receiving `AUTOFILL_CONNECTION_STATUS_CONNECTED` in `connection_status_changed_cb()`.
+
+## Requesting to get the presence of autofill data and autofill authentication information
+
+1. Create the callback function to get the presence of autofill data and autofill authentication information:
+
+   You have to implement the UI to display the name, service message, and logo image of autofill service if the authentication is necessary.
+   In addition, `autofill_auth_info_request()` to retrieve the data previously entered by the user and stored in an autofill service should be called when pressing UI to show autofill service information.
+
+    ```c
+    static void
+    auth_info_received_cb(autofill_h ah, autofill_auth_info_h auth_info_h, void *data)
+    {
+        bool autofill_data_present = false;
+        bool authentication_needed = false;
+        char *service_name = NULL;
+        char *service_message = NULL;
+        char *service_logo_image_path = NULL;
+        char *view_id = NULL;
+
+        autofill_auth_info_get_view_id(auth_info_h, &view_id);
+        autofill_auth_info_get_autofill_data_present(auth_info_h, &autofill_data_present);
+        autofill_auth_info_get_authentication_needed(auth_info_h, &authentication_needed);
+
+        dlog_print(DLOG_INFO, LOG_TAG, "view id : %s, exist autofill data : %d, need authentication : %d", view_id, autofill_data_present, authentication_needed);
+
+        if (view_id)
+            free(view_id);
+
+        if (!autofill_data_present)
+            return;
+
+        if (authentication_needed)
+        {
+            autofill_auth_info_get_service_name(auth_info_h, &service_name);
+            autofill_auth_info_get_service_message(auth_info_h, &service_message);
+            autofill_auth_info_get_service_logo_image_path(auth_info_h, &service_logo_image_path);
+
+            dlog_print(DLOG_INFO, LOG_TAG, "service name : %s, logo path : %s, message : '%s'", service_name, service_logo_image_path, service_message);
+
+            /* create authentication popup */
+            ...
+
+            if (service_message)
+                free(service_message);
+
+            if (service_name)
+                free(service_name);
+
+            if (service_logo_image_path)
+                free(service_logo_image_path);
+        }
+    }
+
+    /* Set callback function for receiving authentication information */
+    autofill_auth_info_set_received_cb(ah, auth_info_received_cb, NULL);
+    ```
+
+2. Create autofill data for each input field and view information to include several autofill data:
```

Comment:
To include several Autofill data, create autofill data for each input field and view information:

Reply:
fixed